public/src/components/channelManagement/bannerTests/bannerUiSelector.tsx

```diff
@@ -0,0 +1,189 @@
+  {
+    template: BannerTemplate.UkraineMomentBanner,
+    label: 'Ukraine Moment Banner 2023',
+  },
+  {
+    template: BannerTemplate.ChoiceCardsBannerBlue,
+    label: 'Choice cards banner - TABS',
+  },
+  {
+    template: BannerTemplate.ChoiceCardsButtonsBannerBlue,
+    label: 'Choice cards banner - BUTTONS',
+  },
+  {
+    template: BannerTemplate.WorldPressFreedomDayBanner,
+    label: 'World Press Freedom Day',
+  },
+  {
+    template: BannerTemplate.Scotus2023MomentBanner,
+    label: 'US Supreme Court 2023 Moment',
+  },
+];
+
+interface BannerTemplateSelectorProps {
+  template: BannerTemplate;
+  onUiChange: (updatedUi: BannerUi) => void;
+  editMode: boolean;
+}
+
+const BannerTemplateSelector: React.FC<BannerTemplateSelectorProps> = ({
+  template,
+  onUiChange,
+  editMode,
+}: BannerTemplateSelectorProps) => {
+  const onChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const value = event.target.value as string;
+    if (isBannerTemplate(value)) {
+      onUiChange(value);
+    }
+  };
+
+  return (
+    <Select value={template} onChange={onChange} disabled={!editMode}>
+      {templatesWithLabels.map(withLabel => (
+        <MenuItem value={withLabel.template} key={withLabel.template}>
+          {withLabel.label}
+        </MenuItem>
+      ))}
+    </Select>
+  );
+};
+
+interface BannerDesignSelectorProps {
+  design: BannerDesignName;
+  onUiChange: (updatedUi: BannerUi) => void;
+  editMode: boolean;
+  designs: BannerDesign[];
+}
+
+const BannerDesignSelector: React.FC<BannerDesignSelectorProps> = ({
+  design,
+  onUiChange,
+  editMode,
+  designs,
+}: BannerDesignSelectorProps) => {
+  const onChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const designName = event.target.value as string;
+    const isValidBannerDesign = designs.map(d => d.name).includes(designName);
+    if (isValidBannerDesign) {
+      onUiChange({ designName });
+    }
+  };
+
+  return (
+    <Select value={design.designName} onChange={onChange} disabled={!editMode}>
+      {designs.map(design => (
+        <MenuItem value={design.name} key={design.name}>
+          {design.name}
+        </MenuItem>
+      ))}
+    </Select>
+  );
+};
+
+type UiType = 'Design' | 'Template';
+
+const BannerUiSelector: React.FC<BannerUiSelectorProps> = ({
+  ui,
+  onUiChange,
+  editMode,
+  designs,
+}: BannerUiSelectorProps) => {
+  const [uiType, setUiType] = useState<UiType>(uiIsDesign(ui) ? 'Design' : 'Template');
+
+  const onUiTypeChange = (event: React.ChangeEvent<{ value: unknown }>): void => {
+    const uiType = event.target.value as UiType;
+    setUiType(uiType);
+
+    onUiChange(
+      uiType === 'Design' ? { designName: 'EXAMPLE' } : BannerTemplate.ContributionsBanner,
+    );
+  };
+
+  useEffect(() => {
+    // This isn't part of the design, so when changes are discarded we want to reflect the unedited configuration
```

Comment:
not sure I understand the meaning of "This isn't part of the design" - can you explain?

Reply:
👍 makes sense